cf-oscore/src/test/java/org/eclipse/californium/oscore/OSCoreObserveTest.java

```diff
@@ -0,0 +1,348 @@
+ ******************************************************************************/
+package org.eclipse.californium.oscore;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+import java.util.Random;
+import java.util.Timer;
+import java.util.TimerTask;
+
+import org.eclipse.californium.category.Large;
+import org.eclipse.californium.core.CoapClient;
+import org.eclipse.californium.core.CoapHandler;
+import org.eclipse.californium.core.CoapObserveRelation;
+import org.eclipse.californium.core.CoapResource;
+import org.eclipse.californium.core.CoapResponse;
+import org.eclipse.californium.core.CoapServer;
+import org.eclipse.californium.core.coap.Request;
+import org.eclipse.californium.core.coap.Response;
+import org.eclipse.californium.core.coap.CoAP.Code;
+import org.eclipse.californium.core.coap.CoAP.ResponseCode;
+import org.eclipse.californium.core.coap.CoAP.Type;
+import org.eclipse.californium.core.coap.MediaTypeRegistry;
+import org.eclipse.californium.core.network.CoapEndpoint;
+import org.eclipse.californium.core.network.EndpointManager;
+import org.eclipse.californium.core.server.resources.CoapExchange;
+import org.eclipse.californium.cose.AlgorithmID;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+
+/**
+ * Performs tests of Observe message exchanges between an OSCORE server and client
+ * Based on interoperability test specification created by Ericsson:
+ * https://ericssonresearch.github.io/OSCOAP/test-spec5.html
+ * 
+ */
+@Category(Large.class)
+public class OSCoreObserveTest {
+	private CoapServer server;
+	private int serverPort;
+	private static String serverAddress = InetAddress.getLoopbackAddress().getHostAddress();
+	
+	private static String clientAddress = InetAddress.getLoopbackAddress().getHostName();
+	
+	private static boolean withOSCORE = true;
+	
+	//OSCORE context information shared between server and client
+	private final static HashMapCtxDB db = HashMapCtxDB.getInstance();
+	private final static AlgorithmID alg = AlgorithmID.AES_CCM_16_64_128;
+	private final static AlgorithmID kdf = AlgorithmID.HKDF_HMAC_SHA_256;
+	private final static byte[] master_secret = { 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B,
+			0x0C, 0x0D, 0x0E, 0x0F, 0x10 };
+	private final static byte[] master_salt = { (byte) 0x9e, (byte) 0x7c, (byte) 0xa9, (byte) 0x22, (byte) 0x23,
+			(byte) 0x78, (byte) 0x63, (byte) 0x40 };
+	
+	@Before
+	public void initLogger() {
+		System.out.println(System.lineSeparator() + "Start " + getClass().getSimpleName());
+		EndpointManager.clear();
+	}
+	
+	//Use the OSCORE stack factory
+	@BeforeClass
+	public static void setStackFactory() {
+		OSCoreCoapStackFactory.useAsDefault();
```

Comment:
Executing the test-suite with maven fails.

`OSCoreCoapStackFactory.useAsDefault();` calls `CoapEndpoint.setDefaultCoapStackFactory(new OSCoreCoapStackFactory());` which must be called only once!.

I would got for 
```
public class OSCoreCoapStackFactory implements CoapStackFactory {
	private static AtomicBoolean init = new AtomicBoolean();
...
	public static void useAsDefault() {
		if (init.compareAndSet(false, true)) {
			CoapEndpoint.setDefaultCoapStackFactory(new OSCoreCoapStackFactory());
		}
	}
}
```

Reply:
Thanks, I have now changed the code according to your suggestion. Done in commit a057caa93983e255fdf48050c820c7902415dbd5.